destination/config.go

```diff
@@ -21,10 +21,7 @@ type Config struct {
 	// Config includes parameters that are the same in the source and destination.
 	common.Config
 
-	// UseSingleShard is a boolean that determines whether to add records in batches using KinesisClient.PutRecords
-	// (ordering not guaranteed, records written to multiple shards)
-	// or to add records to a single shard using KinesisClient.PutRecord
-	// (preserves ordering, records written to a single shard)
-	// Defaults to false to take advantage of batching performance
-	UseSingleShard bool `json:"useSingleShard" validate:"required" default:"true"`
+	// PartitionKey represents the kinesis partition key. Use this to group
+	// records into one or multiple shards.
+	PartitionKey string `json:"partitionKey"`
```

Comment:
This only lets me specify a static partition key, which is not really useful, since whatever I specify here will make records land in the same shard.

What would be useful however is if I could specify a [Go template](https://pkg.go.dev/text/template) here which is evaluated on each record. Then I can specify whatever field I want, or even a combination of fields. By default it would contain `{{ .Key.Bytes }}`, which is the current behavior.